tests/tools/combine-tools.test.js

```diff
@@ -0,0 +1,266 @@
+    { name: 'category2', description: 'Sample Category 2' }
+  ]
+}));
+
+const readJSON = (filePath) => JSON.parse(fs.readFileSync(filePath, 'utf-8'));
+
+describe('combineTools function', () => {
+  const toolsPath = path.join(__dirname, '../', 'fixtures', 'tools', 'tools.json');
+  const tagsPath = path.join(__dirname, '../', 'fixtures', 'tools', 'tags.json');
+  const manualToolsPath = path.join(__dirname, '../', 'fixtures', 'tools', 'manual-tools.json');
+  const automatedToolsPath = path.join(__dirname, '../', 'fixtures', 'tools', 'automated-tools.json');
+
+  let manualTools;
+  let automatedTools;
+  let consoleErrorMock;
+
+  beforeAll(() => {
+    manualTools = readJSON(manualToolsPath);
+    automatedTools = readJSON(automatedToolsPath);
+  });
+
+  afterAll(() => {
+    if (fs.existsSync(toolsPath)) fs.unlinkSync(toolsPath);
+    if (fs.existsSync(tagsPath)) fs.unlinkSync(tagsPath);
+
+    consoleErrorMock.mockRestore();
+  });
+
+  beforeEach(() => {
+    consoleErrorMock = jest.spyOn(console, 'error').mockImplementation(() => { });
+  });
+
+  afterEach(() => {
+    jest.clearAllMocks();
+  });
+
+  it('should combine tools and create correct JSON files', async () => {
+    await combineTools(automatedTools, manualTools, toolsPath, tagsPath);
+
+    const combinedTools = readJSON(toolsPath);
+    expect(combinedTools).toHaveProperty('category1');
+
+    const tagsData = readJSON(tagsPath);
+    expect(tagsData).toHaveProperty('languages');
+    expect(tagsData).toHaveProperty('technologies');
+    expect(tagsData).toEqual(expectedDataT1)
+  });
+
+  it('should handle tools with missing language or technology', async () => {
+
+    await combineTools({}, manualToolsWithMissingData, toolsPath, tagsPath);
+
+    const combinedTools = readJSON(toolsPath);
+    expect(combinedTools).toHaveProperty('category1');
+  });
+
+  it('should sort tools alphabetically by title', async () => {
+
+    await combineTools(manualToolsToSort, {}, toolsPath, tagsPath);
+
+    const combinedTools = readJSON(toolsPath);
+    const toolTitles = combinedTools.category1.toolsList.map(tool => tool.title);
+    expect(toolTitles).toEqual(['Tool A', 'Tool Z']);
+  });
+
+  it('should log validation errors to console.error', async () => {
+
+    await combineTools(automatedToolsT4, manualToolsT4, toolsPath, tagsPath);
+
+    const errorCalls = console.error.mock.calls;
+
+    expect(errorCalls[0][0]).toBe('Script is not failing, it is just dropping errors for further investigation');
+    expect(errorCalls[1][0]).toBe('Invalid Invalid Tool .asyncapi-tool file.');
+    expect(errorCalls[2][0]).toBe('Located in manual-tools.json file');
+    expect(errorCalls[3][0]).toEqual(expect.stringContaining('Validation errors:'));
+
+    expect(fs.existsSync(toolsPath)).toBe(true);
+    expect(fs.existsSync(tagsPath)).toBe(true);
+  });
+
+  it('should handle tools with multiple languages, including new ones', async () => {
+
```

Comment:
Remove extra line from here in every test. Is it due to eslint?

Reply:
Nope, I added that myself. I'll remove it